WordPress/src/main/java/org/wordpress/android/ui/posts/PostSettingsTagsActivity.java

```diff
@@ -0,0 +1,263 @@
+            finish();
+            return;
+        }
+
+        setContentView(R.layout.post_settings_tags_fragment);
+
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setHomeButtonEnabled(true);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.tags_suggestion_list);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        mAdapter = new TagsRecyclerViewAdapter(this);
+        mAdapter.setAllTags(mTaxonomyStore.getTagsForSite(mSite));
+        recyclerView.setAdapter(mAdapter);
+
+        mTagsEditText = (EditText) findViewById(R.id.tags_edit_text);
+        mTagsEditText.addTextChangedListener(this);
+        if (!TextUtils.isEmpty(tags)) {
+            // add a , at the end so the user can start typing a new tag
+            tags += ",";
+            mTagsEditText.setText(tags);
+            mTagsEditText.setSelection(mTagsEditText.length());
+        }
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        mDispatcher.register(this);
+    }
+
+    @Override
+    public void onStop() {
+        mDispatcher.unregister(this);
+        super.onStop();
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putSerializable(WordPress.SITE, mSite);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        int itemId = item.getItemId();
+        if (itemId == android.R.id.home) {
+            saveAndFinish();
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onBackPressed() {
+        saveAndFinish();
+        super.onBackPressed();
+    }
+
+    private void saveAndFinish() {
+        Bundle bundle = new Bundle();
+        bundle.putString(KEY_SELECTED_TAGS, mTagsEditText.getText().toString());
+        Intent intent = new Intent();
+        intent.putExtras(bundle);
+        setResult(RESULT_OK, intent);
+        finish();
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+        // No-op
+    }
+
+    @Override
+    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+        filterListForCurrentText();
+    }
+
+    @Override
+    public void afterTextChanged(Editable editable) {
+        // No-op
+    }
+
+    // Find the text after the last occurrence of "," and filter with it
+    private void filterListForCurrentText() {
+        String text = mTagsEditText.getText().toString();
+        int endIndex = text.lastIndexOf(",");
+        if (endIndex == -1) {
+            mAdapter.filter(text);
+        } else {
+            String textToFilter = text.substring(endIndex + 1);
```

Comment:
I was planning on asking @mattmiklic about that for the v2 as @tonyr59h already mentioned it https://github.com/wordpress-mobile/WordPress-Android/issues/5921#issuecomment-305067327. I made the change in 99da4042623a7814ca70cf1fef73d10398c808cf, however it feels a little weird if the space is added after some text and the suggestions stay the same.